### PR TITLE
Decrease wordiness of string comparisons

### PR DESCRIPTION
**Pull request by clumens**
Here's a couple new functions to cut down on some verbose string comparison blocks.  It's entirely possible I missed a couple - we do an awful lot of string comparisons.

### CODE REVIEW

File: include/crm/common/util.h [hunk 1]
```diff
@@ -62,6 +62,9 @@ char *crm_strdup_printf(char const *format, ...) __attribute__ ((__format__ (__p
 int pcmk__parse_ll_range(const char *srcstring, long long *start, long long *end);
 gboolean pcmk__str_in_list(GList *lst, const gchar *s);
 
+bool pcmk_str_any_of(const char *s, ...) G_GNUC_NULL_TERMINATED;
+bool pcmk_str_none_of(const char *s, ...) G_GNUC_NULL_TERMINATED;
```
Comment (kgaillot) [inline include/crm/common/util.h:66]: I'd rather these be internal (in include/crm/common/internal.h). As Beekhof once put it to me, "we don't want libcrmcommon to become another glib". If something's not cluster-related or required to use something cluster-related, we should prefer internal to keep the flexibility to change at will.

(Of course a bunch of old functions are public that should be internal by this principle, but we have to wait until the next API compatibility break to do that)

File: lib/pengine/utils.c [hunk 2]
```diff
@@ -808,10 +806,9 @@ unpack_operation_on_fail(pe_action_t * action)
         interval_spec = crm_element_value(action->op_entry,
                                           XML_LRM_ATTR_INTERVAL);
 
-        if (safe_str_neq(name, CRMD_ACTION_PROMOTE)
-            && (safe_str_neq(name, CRMD_ACTION_STATUS)
+        if (pcmk_str_none_of(name, CRMD_ACTION_PROMOTE, CRMD_ACTION_STATUS, NULL)
                 || safe_str_neq(role, "Master")
-                || (crm_parse_interval_spec(interval_spec) == 0))) {
+                || (crm_parse_interval_spec(interval_spec) == 0)) {
```
[the remark this answers is not in the document]
Comment (clumens) [reply lib/pengine/utils.c:811]: Now that is some thorough code review!  Changed.

File: include/crm/common/util.h (hunk 1 above)
Comment (clumens) [reply include/crm/common/util.h:66]: That is a good, concise rule to live by.  Changed.

**Comment by clumens** [comment]
This PR came about when working on converting crm_resource.c to use glib for command line handling.  I just couldn't look at those blocks in the while loop in main without doing something about it.

File: include/crm/common/util.h [hunk 3]
```diff
@@ -62,6 +62,9 @@ char *crm_strdup_printf(char const *format, ...) __attribute__ ((__format__ (__p
 int pcmk__parse_ll_range(const char *srcstring, long long *start, long long *end);
 gboolean pcmk__str_in_list(GList *lst, const gchar *s);
 
+bool pcmk__str_any_of(const char *s, ...) G_GNUC_NULL_TERMINATED;
+bool pcmk__str_none_of(const char *s, ...) G_GNUC_NULL_TERMINATED;
```
Comment (kgaillot) [inline include/crm/common/util.h:66]: Along with making these internal, can you move them (and the two above them) to common/internal.h?

Comment (clumens) [reply include/crm/common/util.h]: Yep, done.  That should have been in the last push in the first place.

**Pull request merged**